store/src/genesis.rs

```diff
@@ -1,21 +1,99 @@
-use miden_crypto::merkle::{EmptySubtreeRoots, MmrPeaks, TieredSmt};
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use miden_crypto::{
+    dsa::rpo_falcon512,
+    merkle::{EmptySubtreeRoots, MerkleError, MmrPeaks, SimpleSmt, TieredSmt},
+    Felt,
+};
+use miden_lib::{faucets::create_basic_fungible_faucet, wallets::create_basic_wallet, AuthScheme};
 use miden_node_proto::block_header;
-use miden_objects::{notes::NOTE_LEAF_DEPTH, Digest};
+use miden_objects::{
+    accounts::Account, assets::TokenSymbol, notes::NOTE_LEAF_DEPTH, BlockHeader, Digest,
+};
+use serde::{Deserialize, Serialize};
 
 use crate::state::ACCOUNT_DB_DEPTH;
 
-/// Generates the header of the genesis block. The timestamp is currently hardcoded to be the UNIX epoch.
-pub fn genesis_header() -> block_header::BlockHeader {
-    block_header::BlockHeader {
-        prev_hash: Some(Digest::default().into()),
-        block_num: 0,
-        chain_root: Some(MmrPeaks::new(0, Vec::new()).unwrap().hash_peaks().into()),
-        account_root: Some(EmptySubtreeRoots::entry(ACCOUNT_DB_DEPTH, 0).into()),
-        nullifier_root: Some(TieredSmt::default().root().into()),
-        note_root: Some(EmptySubtreeRoots::entry(NOTE_LEAF_DEPTH, 0).into()),
-        batch_root: Some(Digest::default().into()),
-        proof_hash: Some(Digest::default().into()),
-        version: 0,
-        timestamp: 0,
+/// Represents the state at genesis, which will be used to derive the genesis block.
+#[derive(Serialize, Deserialize)]
+pub struct GenesisState {
+    pub accounts: Vec<Account>,
+    pub version: u64,
+    pub timestamp: u64,
+}
+
+impl GenesisState {
+    pub fn new(pub_key: rpo_falcon512::PublicKey) -> Self {
+        let accounts = {
+            let mut accounts = Vec::new();
```

Comment:
I would probably move the construction logic outside of the `GenesisState` struct. Basically, the constructor could be:
```Rust
pub fn new(accounts: Vec<Account>, version: u64, timestamp: u64) -> Self
```
This would let us create `GenesisState` in more flexible ways in the future.